### PR TITLE
fix: convert async tests to synchronous

### LINKED ISSUE #38254: Slow react tests
**Describe your problem and how to reproduce it:**
Several React challenges have asynchronous tests that include numerous timeouts.  This can cause `npm run test` to fail randomly causing ci tests to, mistakenly, flag a commit as failing when it should not.  Most recently curriculum/challenges/english/03-front-end-libraries/react/add-event-listeners.english.md failed, but it may not be the only one.

I don't think there is a need to wait, since the testing framework is synchronous. So, they should be re-written as synchronous tests.

Reproduction is difficult, but it should be fixed if the tests run quickly (i.e under a second).

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38254 

This PR prevents `npm run test` from accidentally failing due to asynchronous tests causing a timeout

<s>There are 2 tests which still take `900 - 1000` ms to complete which cannot be made synchronous, because they rely on the `setState` method.</s>
- Pass a Callback as Props
- Set State with this.setState

### CODE REVIEW

**Comment by Sky020** [comment]
@ojeytonwilliams This challenge and test appears to be slightly different to all of the others:

<pre>
  Check challenges
    Check challenges (english)
      React
        Access Props Using this.props
          ✓ Matches a title in meta.json
          ✓ Matches an ID in meta.json
          ✓ Common checks
⠼ Testing          ✓ Test suite must fail on the initial contents (3284ms)
          Check tests syntax
            ✓ Check for: The <code>ResetPassword</code> component should return a single <code>div</code> element.
            ✓ Check for: The fourth child of <code>ResetPassword</code> should be the <code>ReturnTempPassword</code> component.
            ✓ Check for: The <code>ReturnTempPassword</code> component should have a prop called <code>tempPassword</code>.
            ✓ Check for: The <code>tempPassword</code> prop of <code>ReturnTempPassword</code> should be equal to a string of at least <code>8</code> characters.
            ✓ Check for: The <code>ReturnTempPassword</code> component should display the password you create as the <code>tempPassword</code> prop within <code>strong</code> tags.
          Check tests against solutions
⠧ Testing            ✓ Solution 1 must pass the tests (251ms)
</pre>

The _Test suite must fail on the initial contents_ takes 20-30 times longer to complete, when compared to the following challenges. Is this something that needs to be further looked into? Does it just have to do with the fact that it is the first challenge in the block?

**Review by moT01 — DISMISSED** [review]
LGTM  🎉

File: curriculum/challenges/english/03-front-end-libraries/react/create-a-controlled-form.english.md [hunk 1]
```diff
@@ -63,44 +126,43 @@ class MyForm extends React.Component {
   }
   handleSubmit(event) {
     // change code below this line
-
```
Comment (ojeytonwilliams) [inline curriculum/challenges/english/03-front-end-libraries/react/create-a-controlled-form.english.md:66]: Could you put this back?  It looks like eslint/prettier was a bit overzealous.

**Comment by ojeytonwilliams** [comment]
> Does it just have to do with the fact that it is the first challenge in the block?

At a guess, yes, it's probably loading Enzyme and that's delaying it.  There's nothing obviously wrong with the tests.  

I don't think we need to worry about it yet. The other changes should be a big help and we know what to look for if the tests randomly fail in future.

**Review by moT01 — APPROVED** [review]
Still LGTM 🎉

**Review by ojeytonwilliams — APPROVED** [review]
LGTM too 👍

**Pull request merged**